src/openrct2/peep/Staff.cpp

```diff
@@ -604,49 +603,49 @@ static int32_t staff_handyman_direction_rand_surface(Peep* peep, uint8_t validDi
     // If it tries all directions this is required
     // to make it back to the first direction and
     // override validDirections
-    direction &= 3;
-    return direction;
+    newDirection %= NumOrthogonalDirections;
```

Comment:
Why the change from & to %?

Reply:
Because that's what the `& 3` does. But `% NumOrthogonalDirections` is much clearer, since it shows that we're truncating the direction to a valid one.

Reply:
Bare in mind that unless `newDirection` is unsigned, it will not be as efficient / generate the same instructions as `&`.

Reply:
Good point. In this case, though, it uses an unsigned 8-bit integer.